src/h3lib/lib/polyfill.c

```diff
@@ -647,5 +678,66 @@ H3Error H3_EXPORT(polygonToCellsExperimental)(const GeoPolygon *polygon,
     for (; iter.cell; iterStepPolygon(&iter)) {
         out[i++] = iter.cell;
     }
+    return iter.error;
+}
+
+static int MAX_SIZE_CELL_THRESHOLD = 10;
```

Comment:
Per @dfellis - we could potentially allow the caller to configure this value, by supplying it in the flags. The caller could give either a specific value, or an "order of magitude" value (exponent of 10) that could make this more accurate.